backend/app/mailers/devise_mailer.rb

```diff
@@ -0,0 +1,25 @@
+class DeviseMailer < Devise::Mailer
+  def confirmation_instructions(record, token, opts = {})
+    @token = token
+
+    I18n.with_locale(record.locale || opts[:fallback_language]) do
+      devise_mail(record, :confirmation_instructions, opts)
+    end
+  end
+
+  def reset_password_instructions(record, token, opts = {})
+    @token = token
+
+    I18n.with_locale(record.locale || opts[:fallback_language]) do
+      devise_mail(record, :reset_password_instructions, opts)
+    end
+  end
+
+  def invitation_instructions(record, token, opts = {})
+    @token = token
+
+    I18n.with_locale(record.locale || opts[:fallback_language]) do
```

Comment:
Unfortunately, this does not work. Your idea is correct, but It would work only if we have used `deliver_now` at `send_devise_notification` so email is send immediately and takes into account `I18n.with_locale`. I have switched to `deliver_later` which pushes mail to queue and when it gets picked up by different process/server, It does not remember what was original locale before it got enqueued --> that is reason why localization needs to be put directly to mailer.

Reply:
Ahh, ok, right.